File: internal/model/asset.go
package model

import "time"

// Asset 资产表
type Asset struct {
	ID              uint      `gorm:"primaryKey" json:"id"`
	Code            string    `gorm:"index;default:'';not null" json:"code"` // 基金代码，如"050027"
	Name            string    `gorm:"default:'';not null" json:"name"`       // 资产名称，如"博时信用债纯债债券A"
	URL             string    `gorm:"default:''" json:"url"`                 // 基金详情页 URL
	Type            string    `gorm:"index;not null" json:"type"`            // stock/bond
	Source          string    `gorm:"not null" json:"source"`                // 支付宝/天天基金
	EncryptedAmount string    `gorm:"type:text;not null" json:"-"`           // 加密后的金额
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

// 资产类型常量
const (
	AssetTypeStock = "stock"
	AssetTypeBond  = "bond"
)

File: lib/data/models/asset.dart
/// 资产类型枚举
enum AssetType {
  stock('stock', '股票型'),
  bond('bond', '债券型');

  const AssetType(this.value, this.label);
  final String value;
  final String label;

  static AssetType fromString(String value) {
    return AssetType.values.firstWhere(
      (type) => type.value == value,
      orElse: () => AssetType.stock,
    );
  }
}

/// 资产模型
class Asset {
  final int? id;
  final String code;
  final String name;
  final String url;
  final AssetType type;
  final String source;
  final String encryptedAmount;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Asset({
    this.id,
    required this.code,
    required this.name,
    required this.url,
    required this.type,
    required this.source,
    required this.encryptedAmount,
    required this.createdAt,
    required this.updatedAt,
  });

  /// 从数据库 Map 创建
  factory Asset.fromMap(Map<String, dynamic> map) {
    return Asset(
      id: map['id'] as int?,
      code: map['code'] as String,
      name: map['name'] as String,
      url: map['url'] as String,
      type: AssetType.fromString(map['type'] as String),
      source: map['source'] as String,
      encryptedAmount: map['encrypted_amount'] as String,
      createdAt: DateTime.parse(map['created_at'] as String),
      updatedAt: DateTime.parse(map['updated_at'] as String),
    );
  }

  /// 转换为数据库 Map
  Map<String, dynamic> toMap() {
    return {
      if (id != null) 'id': id,
      'code': code,
      'name': name,
      'url': url,
      'type': type.value,
      'source': source,
      'encrypted_amount': encryptedAmount,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }

  /// 复制并修改部分字段
  Asset copyWith({
    int? id,
    String? code,
    String? name,
    String? url,
    AssetType? type,
    String? source,
    String? encryptedAmount,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Asset(
      id: id ?? this.id,
      code: code ?? this.code,
      name: name ?? this.name,
      url: url ?? this.url,
      type: type ?? this.type,
      source: source ?? this.source,
      encryptedAmount: encryptedAmount ?? this.encryptedAmount,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
